### PR TITLE
chore: add rule and test

### PR DESCRIPTION
**Pull request by caicancai**
I noticed that risinglight uses egg for CBO optimization, but it looks violent at the moment. I will try to optimize it.

### CODE REVIEW

**Comment by caicancai** [comment]
cc @wangrunji0408 If you have time, could you help me review this PR? Thank you.

File: src/planner/rules/plan.rs [hunk 1]
```diff
@@ -582,6 +583,47 @@ mod tests {
...
+        merge_filter,
+        rules(),
+        // SELECT name
+        // FROM student
+        // LIMIT 10
+        // Order by name
+        "
+        proj (list $1.2)
+        (limit 10 0
+            (order (list)
```
Comment (wangrunji0408) [inline src/planner/rules/plan.rs:616]: This is an empty ordering. Should be `(order (list $1.2) ..)`?

Comment (caicancai) [reply src/planner/rules/plan.rs]: Thanks, fix
But the fact that rust test can pass means there is still room for improvement. I will find time to improve it.

**Comment by caicancai** [comment]
@wangrunji0408 PATL，thank you

File: src/planner/rules/plan.rs [hunk 2]
```diff
@@ -582,6 +583,47 @@ mod tests {
...
+        // Order by name
+        "
+        proj (list $1.2)
+        (limit 10 0
+            (order (list $1.2) 
+                (scan $1 (list $1.1 $1.2) null)
+            )
+        )" => "
+        proj (list $1.2)
+        （topn 10 0 (list)
```
Comment (wangrunji0408) [inline src/planner/rules/plan.rs:621]: But why this test can pass?

```suggestion
        （topn 10 0 (list $1.2) 
```

Comment (caicancai) [reply src/planner/rules/plan.rs:621]: I think this is a bug and I will fix it

**Comment by caicancai** [comment]
@wangrunji0408 I would like to ask a question, can egg+sql optimization be extracted into a separate framework and then integrated with calcite? Do you think this is feasible?

File: src/planner/rules/plan.rs (hunk 2 above)
Comment (caicancai) [reply src/planner/rules/plan.rs]: cli failed.
Why?

Comment (wangrunji0408) [reply src/planner/rules/plan.rs]: Permission denied. It's a random happened failure. Let's retry it.

**Comment by wangrunji0408** [comment]
> can egg+sql optimization be extracted into a separate framework and then integrated with calcite? Do you think this is feasible?

I'm not sure as the Calcite is written in Java while egg is designed for Rust. Besides, what's the purpose of doing this? As far as I know, the Calcite itself is a framework for optimization. I have always thought of egg as a Rust alternative of Calcite.

**Comment by caicancai** [comment]
> I'm not sure as the Calcite is written in Java while egg is designed for Rust. Besides, what's the purpose of doing this? As far as I know, the Calcite itself is a framework for optimization. I have always thought of egg as a Rust alternative of Calcite.

I think it is a bit complicated to expand an optimization rule in Calcite at present, and it is not as convenient as egg. My original intention is to make Calcite more convenient to use.

**Review by wangrunji0408 — APPROVED** [review]
Thanks!

**Pull request merged**